client/components/productForm.js

```diff
@@ -0,0 +1,94 @@
+import React, {Component} from 'react'
+import {Form} from 'semantic-ui-react'
+import {connect} from 'react-redux'
+import {addNewProduct, editOneProduct} from '../store/products'
+
+class ProductForm extends Component {
+  state = {
+    name: '',
+    description: '',
+    stock: 0,
+    price: 0
+  }
+
+  componentDidMount() {
+    if (this.props.product) {
+      const {name, description, stock, price} = this.props.product
+      this.setState({name, description, stock, price})
+    }
+  }
+
+  handleChange = evt => this.setState({[evt.target.name]: evt.target.value})
+
+  handleSubmit = evt => {
+    evt.preventDefault()
+    const {name, description, stock, price} = this.state
+    const product = {name, description, stock, price}
+    if (this.props.product) {
+      this.props.editProduct(product)
+    } else {
+      this.props.newProduct(product)
+    }
+    this.setState({
```

Comment:
I'm thinking for the Edit Product, we could set the state based on the selectedProduct in the store to let the user know what the previous values were